Battle Transitions

ificiana opened the pull request:
- [x] make the transition into battle
- [ ] return to map transition
- [ ] (enhancement) implement a better approach using event managers

ificiana committed:
make transition into battle

ificiana committed:
restructure helpers as utils

ificiana committed:
move cache into utils

ificiana committed:
change views API, add an eventhandler

ificiana committed:
incorporate the eventhandler

ificiana committed:
rework buttons

VagishVela reviewed (dismissed):
Let's merge then fix bugs in dev in this case... moving fast💨

ificiana committed:
merge 'dev' into 'transitions'

VagishVela committed:
fix removed button code from merge

This adds back a loose check to see if a mouseclick happened. The bug existing before where it doesn't register every click is still there.
